src/Symfony/Component/Security/Http/Firewall/UsernamePasswordJsonAuthenticationListener.php

```diff
@@ -117,6 +118,10 @@ public function handle(GetResponseEvent $event)
             $response = $this->onFailure($request, $e);
         }
 
+        if (null === $response) {
```

Comment:
In case of success, the listener just lets the original request succeeds so it doesn't return any response